ResearchKit/Common/ORKDiscreteGraphView.m

```diff
@@ -0,0 +1,1023 @@
+
+static CGFloat const kFadeAnimationDuration = 0.2;
+static CGFloat const kGrowAnimationDuration = 0.1;
+static CGFloat const kPopAnimationDuration  = 0.3;
+
+static CGFloat const kSnappingClosenessFactor = 0.3f;
+
+@interface ORKDiscreteGraphView ()
+
+@property (nonatomic, strong) NSMutableArray *dataPoints;//actual data
+
+@property (nonatomic, strong) NSMutableArray *xAxisPoints;
+
+@property (nonatomic, strong) NSMutableArray *yAxisPoints;//normalised for this view
+
+@property (nonatomic, strong) UIView *plotsView; //Holds the plots
+
+@property (nonatomic, strong) ORKAxisView *xAxisView;
+
+@property (nonatomic, strong) UIView *yAxisView;
+
+@property (nonatomic, strong) UILabel *emptyLabel;
+
+@property (nonatomic) BOOL hasDataPoint;
+
+@property (nonatomic, strong) UIView *scrubberLine;
+
+@property (nonatomic, strong) UILabel *scrubberLabel;
+
+@property (nonatomic, strong) UIView *scrubberThumbView;
+
+@property (nonatomic, readwrite) CGFloat minimumValue;
+
+@property (nonatomic, readwrite) CGFloat maximumValue;
+
+@property (nonatomic, strong) NSMutableArray *xAxisTitles;
+
+@property (nonatomic) NSInteger numberOfXAxisTitles;
+
+@property (nonatomic, strong) NSMutableArray *referenceLines;
+
+@property (nonatomic, strong) NSMutableArray *pathLines;
+
+@property (nonatomic, strong) NSMutableArray *dots;
+
+@property (nonatomic) BOOL shouldAnimate;
+
+@end
+
+
+@implementation ORKDiscreteGraphView
+
+@synthesize tintColor = _tintColor;
+@synthesize maximumValue = _maximumValue;
+@synthesize minimumValue = _minimumValue;
+
+#pragma mark - Init
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    if (self = [super initWithFrame:frame]) {
+    }
+    return self;
+}
+
+- (id)initWithCoder:(NSCoder *)aDecoder {
+    if (self = [super initWithCoder:aDecoder]) {
+    }
+    return self;
+}
+
+- (void)sharedInit {
+    [super sharedInit];
+    
+    _dataPoints = [NSMutableArray new];
+    
+    _xAxisPoints = [NSMutableArray new];
+    _yAxisPoints = [NSMutableArray new];
+    
+    _xAxisTitles = [NSMutableArray new];
+    
+    _referenceLines = [NSMutableArray new];
+    _pathLines = [NSMutableArray new];
+    _dots = [NSMutableArray new];
+    
+    _tintColor = [UIColor colorWithRed:244/255.f green:190/255.f blue:74/255.f alpha:1.f];
+    
+    _hasDataPoint = NO;
+    _shouldAnimate = YES;
+    _shouldConnectRanges = YES;
+    
+    [self setupViews];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(animateLayersSequentially) name:kORKDiscreteGraphViewTriggerAnimationsNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(refreshGraph) name:kORKDiscreteGraphViewRefreshNotification object:nil];
+}
+
+- (void)setupViews {
+    /* ----------------- */
+    /* Basic Views */
+    /* ----------------- */
+    
+    _plotsView = [UIView new];
+    _plotsView.backgroundColor = [UIColor clearColor];
```

Comment:
Maybe define a `cachedColorMethod(ork_clearColor, 0.0, 0.0, 0.0., 0.0)` in `ORKSkin`?